earthfile2llb/earthfile2llb.go

```diff
@@ -105,18 +111,19 @@ func Earthfile2LLB(ctx context.Context, target domain.Target, opt ConvertOpt) (m
 		return nil, errors.Wrapf(err, "resolve build context for target %s", target.String())
 	}
 
-	ftrs, err := features.GetFeatures(bc.Earthfile.Version)
+	ftrs, err := features.GetFeatures(bc.Earthfile.Version) // TODO ACB how do I match this up with builder/builder.go:334 on performSaveLocals
 	if err != nil {
 		return nil, errors.Wrapf(err, "resolve feature set for version %v for target %s", bc.Earthfile.Version.Args, target.String())
 	}
-
 	err = features.ApplyFlagOverrides(ftrs, opt.FeatureFlagOverrides)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to apply version feature overrides")
 	}
+	opt.Features = ftrs
 
-	if ftrs.ReferencedSaveOnly {
-		fmt.Printf("TODO feature-flip referenced save artifact as local feature in a future PR.\n")
+	if !ftrs.ReferencedSaveOnly {
+		// 0.5 behaviour is to save all artifacts and images in all cases (if they are directly referenced, or indirectly triggered)
```

Comment:
Not really.. in 0.5 artifacts are to be saved if they are part of local targets, whereas images are always saved.

Reply:
Good catch, I will expand the tests to make sure the save images still occur.

Reply:
@vladaionescu I just looked over https://github.com/earthly/earthly/issues/896 again and it says:

> We change the logic s.t. output is only produced for the targets invoked either directly (mentioned directly on the command-line) or via calls to BUILD. FROM and COPY (and others e.g. WITH DOCKER --load) do not cause images to be output locally.

Is this still correct? I have this code only locally saving artifacts or images when they are referenced via `FROM` or `BUILD`.